flink-ml-benchmark/src/test/java/org/apache/flink/ml/benchmark/BenchmarkTest.java

```diff
@@ -54,6 +60,44 @@ public void testParseJsonFile() throws Exception {
         assertTrue(benchmarks.containsKey("KMeansModel-1"));
     }
 
+    @Test
+    public void testJsonFileLegality() throws IOException, ClassNotFoundException {
+        File resourcesDir =
+                new File(
+                                this.getClass()
+                                        .getClassLoader()
+                                        .getResource("benchmark-demo.json")
+                                        .getPath())
+                        .getParentFile();
+        File[] jsonFiles =
+                resourcesDir.listFiles(
+                        (dir, name) ->
+                                name.endsWith(".json") && !name.equals("benchmark-demo.json"));
+
+        for (File file : jsonFiles) {
+            Map<String, Map<String, Map<String, ?>>> benchmarks =
+                    BenchmarkUtils.parseJsonFile(file.getAbsolutePath());
+            for (Map<String, Map<String, ?>> params : benchmarks.values()) {
+                assertTrue(
+                        Arrays.asList("stage", "inputData", "modelData")
+                                .containsAll(params.keySet()));
+
+                WithParams stage = ReadWriteUtils.instantiateWithParams(params.get("stage"));
+                assertTrue(stage instanceof Stage);
```

Comment:
It is nice to add this test 👍

Would it be simpler to just copy/paste the following lines from `BenchmarkUtils#runBenchmark` instead using `instanceof`? This approach also reduces the `Raw use of parameterized class ...` warnings. 

```
Stage stage = ReadWriteUtils.instantiateWithParams(params.get("stage"));
InputDataGenerator inputDataGenerator =
        ReadWriteUtils.instantiateWithParams(params.get("inputData"));
DataGenerator modelDataGenerator = null;
if (params.containsKey("modelData")) {
    modelDataGenerator = ReadWriteUtils.instantiateWithParams(params.get("modelData"));
}
```

Reply:
Using `instanceOf` gives a clearer error message if someone writes an incorrect class in json file, instead of `ClassCastException`. The approach in `BenchmarkUtils#runBenchmark` has no warning because of the `@SuppressWarnings` annotation, I can add this annotation.

Reply:
Error message from unit test is supposed be read by developers and it is OK for ask developers to understand and debug ClassCastException in this case, right?

If we want error message to be more meaningful to users who run benchmarks, should we update `ReadWriteUtils#instantiateWithParams()` to catch `ClassCastException` and replace it properly?

Can you provide an example stack trace in the case described above so that we can check whether we need to improve it?

Reply:
An example stack trace is like this:
```
java.lang.ClassCastException: org.apache.flink.ml.feature.imputer.ImputerTest cannot be cast to org.apache.flink.ml.api.Stage
	at org.apache.flink.ml.benchmark.BenchmarkTest.testJsonFileLegality(BenchmarkTest.java:137)
	at org.apache.flink.util.TestNameProvider$1.evaluate(TestNameProvider.java:45)
```

Reply:
Sounds good. Thanks for the example.

Reply:
Sounds good. Thanks for the example.